include/z64environment.h

```diff
@@ -132,17 +132,34 @@ typedef struct {
     /* 0x06 */ u8 skybox2Index;
 } TimeBasedSkyboxEntry; // size = 0x8
 
+// Get blend rate from `EnvLightSettings.blendRateAndFogNear` in 0-255 range
+#define ENV_LIGHT_SETTINGS_BLEND_RATE_U8(blendRateAndFogNear) (((blendRateAndFogNear) >> 10) * 4)
+#define ENV_LIGHT_SETTINGS_FOG_NEAR(blendRateAndFogNear) ((blendRateAndFogNear) & 0x3FF)
+
 typedef struct {
     /* 0x00 */ u8 ambientColor[3];
     /* 0x03 */ s8 light1Dir[3];
     /* 0x06 */ u8 light1Color[3];
     /* 0x09 */ s8 light2Dir[3];
     /* 0x0C */ u8 light2Color[3];
     /* 0x0F */ u8 fogColor[3];
-    /* 0x12 */ s16 fogNear;
-    /* 0x14 */ s16 fogFar;
+    /* 0x12 */ s16 blendRateAndFogNear;
+    /* 0x14 */ s16 zFar;
 } EnvLightSettings; // size = 0x16
 
+typedef EnvLightSettings LightSettings;
+
+typedef struct {
+    /* 0x00 */ u8 ambientColor[3];
+    /* 0x03 */ s8 light1Dir[3];
+    /* 0x06 */ u8 light1Color[3];
+    /* 0x09 */ s8 light2Dir[3];
+    /* 0x0C */ u8 light2Color[3];
+    /* 0x0F */ u8 fogColor[3];
+    /* 0x12 */ s16 fogNear;
+    /* 0x14 */ s16 zFar;
+} CurrentEnvLightSettings; // size = 0x16
+
```

Comment:
it was a conscious choice to not make a new struct, when the only small difference was the blend rate being packed into fogNear.
We can talk about it again, but just saying that it was done on purpose

Reply:
Afaict there is no constraint for `EnvLightSettings` and `CurrentEnvLightSettings` to be the same layout, but I agree they were most likely the same struct initially

Still this seemed to me like a better idea than either
- using `blendRateAndFogNear` as just `fogNear` (not bitpacked) for `EnvironmentContext.lightSettings`
- using `fogNear` as `blendRateAndFogNear` (undocumented bitpacked besides macros and comments) for the scene's `EnvLightSettings` array

Reply:
I would rather it be the same structure and document the blend rate better personally

Reply:
I actually think the blend rate is more important since it defines how quickly the light settings transition from one settings entry to another (when the env system is doing settings blending), which affects all members and not just fog near.
The effect of tweaking the blend rate seems more immediately noticeable to me than the fog distance

Reply:
im not saying that its less important, but overall the fields main purpose is fogNear. Its only the blend rate temporarily.
We will probably just need more opinions, cause I still would like it to be how I originally had it

Reply:
This sounds like a union to me.

Reply:
My opinion is pretty split on this, but in this case I think it makes sense to have separate structs since there isn't any struct copying or anything like that happening, and the combined field only exists in the "scene" struct. It's a bit unfortunate to have an extra struct just for this difference, but it does allow us to name the field properly for the "current" settings, which is the struct used in most places (and externally)

Reply:
I still dont fully agree, but I will concede for the sake of moving forward

Reply:
I could add a comment near `CurrentEnvLightSettings` like 

> Nearly the same struct as `EnvLightSettings`, but `CurrentEnvLightSettings.fogNear` is not a bitfield like `blendRateAndFogNear` is

?

Reply:
Im not sure tbh. On one hand Im fine with it, on the other im wondering if that would just make things more confusing. may be best to just committing to them being different things and letting the reader work out that small difference if they choose to